cdap-kubernetes/src/main/java/co/cask/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -0,0 +1,474 @@
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withBundlerClassAcceptor(ClassAcceptor classAcceptor) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withMaxRetries(String runnableName, int maxRetries) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer addSecureStore(SecureStore secureStore) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setLogLevel(LogEntry.Level logLevel) {
+    return setLogLevels(Collections.singletonMap(Logger.ROOT_LOGGER_NAME, logLevel));
+  }
+
+  @Override
+  public TwillPreparer setLogLevels(Map<String, LogEntry.Level> logLevels) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setLogLevels(String runnableName, Map<String, LogEntry.Level> logLevelsForRunnable) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setClassLoader(String classLoaderClassName) {
+    return this;
+  }
+
+  @Override
+  public TwillController start() {
+    return start(60, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public TwillController start(long timeout, TimeUnit timeoutUnit) {
+    try {
+      V1ConfigMap configMap = createConfigMap();
+      createDeployment(configMap);
+      return new KubeTwillController(resourceMeta.getName(), kubeNamespace, runId, apiClient, discoveryServiceClient);
+    } catch (ApiException | IOException e) {
+      throw new RuntimeException("Unable to create Kubernetes resource while attempting to start program.", e);
+    }
+  }
+
+  private V1Deployment createDeployment(V1ConfigMap configMap) throws ApiException {
+    /*
+       The deployment will look something like:
+       apiVersion: apps/v1
+       kind: Deployment
+       metadata:
+         [deployment meta]
+       spec:
+         [deployment spec]
+       template:
+         [template spec]
+     */
+    AppsV1Api appsApi = new AppsV1Api(apiClient);
+    V1Deployment deployment = new V1Deployment();
+
+    /*
+       metadata:
+         name: service-runid
+         labels:
+           app: cdap
+           cdap-namespace: system
+           cdap-app: dataprep
+           cdap-program-type: SERVICE
+           cdap-program: service
+           cdap-run: runid
+     */
+    deployment.setMetadata(resourceMeta);
+
+    /*
+       spec:
+         replicas: 1
+         selector:
+           matchLabels:
+             app: cdap
+             cdap-namespace: system
+             cdap-app: dataprep
+             cdap-program-type: SERVICE
+             cdap-program: service
+             cdap-run: runid
+     */
+    V1DeploymentSpec spec = new V1DeploymentSpec();
+    V1LabelSelector labelSelector = new V1LabelSelector();
+    labelSelector.matchLabels(resourceMeta.getLabels());
+    spec.setSelector(labelSelector);
+    // TODO: figure out where number of instances is specified in twill spec
+    spec.setReplicas(1);
+
+    /*
+       template:
+         metadata:
+           labels:
+             app: cdap
```

Comment:
For labels, try to follow the operator standard, `cdap.*`. This is also the prefix filtered by the K8s Environment implementation.

Reply:
these are old comments that don't reflect reality, will update.